src/System.Data.Common/tests/System/Data/SqlTypes/SqlDecimalTest.cs

```diff
@@ -180,53 +143,31 @@ public void ArithmeticMethods()
             Assert.Equal(-2006m, SqlDecimal.Add(_test4, test2));
             Assert.Equal(8000.00m, SqlDecimal.Add(test2, _test3));
 
-            try
-            {
-                SqlDecimal test = SqlDecimal.Add(SqlDecimal.MaxValue, SqlDecimal.MaxValue);
-                Assert.False(true);
-            }
-            catch (OverflowException)
-            {
-            }
+            Assert.Throws<OverflowException>(() => SqlDecimal.Add(SqlDecimal.MaxValue, SqlDecimal.MaxValue));
 
             Assert.Equal(6465m, SqlDecimal.Ceiling(_test1));
             Assert.Equal(SqlDecimal.Null, SqlDecimal.Ceiling(SqlDecimal.Null));
 
-            // Divide() => Notworking
-            /*
-            Assert.Equal ((SqlDecimal)(-1077.441066m), SqlDecimal.Divide (Test1, Test4));
-            Assert.Equal (1.54687501546m, SqlDecimal.Divide (Test2, Test1).Value);
+            // Divide
+			Assert.Equal(-1077.441066m, SqlDecimal.Divide(_test1, _test4));
+			Assert.Equal(1.54687501546m, SqlDecimal.Divide(_test2, _test1).Value, 9);
 
-            try {
-                SqlDecimal test = SqlDecimal.Divide(Test1, new SqlDecimal(0)).Value;
-Assert.False(true);
-            } catch (DivideByZeroException e) {
-                Assert.Equal (typeof (DivideByZeroException), e.GetType ());
-            }
-            */
+            Assert.Throws<DivideByZeroException>(() => SqlDecimal.Divide(_test1, new SqlDecimal(0)));
 
             Assert.Equal(6464m, SqlDecimal.Floor(_test1));
 
             // Multiply()
             SqlDecimal Test;
             SqlDecimal test1 = new SqlDecimal(2m);
-            Assert.Equal(64646464.000000m, SqlDecimal.Multiply(_test1, _test2).Value);
+            Assert.Equal(64646464m, SqlDecimal.Multiply(_test1, _test2).Value);
```

Comment:
it.... looked like the trailing zeros had some effect when comparing the results from `ToString`; might want to revert this? don't know what exactly the impact is.